core/src/main/java/brooklyn/entity/basic/EntityPredicates.java

```diff
@@ -271,9 +274,38 @@ public boolean apply(@Nullable Entity input) {
     // ---------------------------
 
     /**
+     * @param typeRegex a regular expression
+     * @return true if any of the interfaces implemented by the entity (including those derived) match typeRegex.
+     */
+    public static Predicate<Entity> hasInterfaceMatching(String typeRegex) {
+        return new ImplementsInterface(typeRegex);
+    }
+
+    protected static class ImplementsInterface implements SerializablePredicate<Entity> {
+        protected final Pattern pattern;
+
+        public ImplementsInterface(String typeRegex) {
+            this.pattern = Pattern.compile(typeRegex);
+        }
+
+        @Override
+        public boolean apply(@Nullable Entity input) {
+            if (input == null) return false;
+            for (Class<?> cls : ClassUtils.getAllInterfaces(input.getClass())) {
```

Comment:
commons-lang3 is coming from camp-base rather than core directly. I'll swap it for Reflections.